HWIOAuthBundle.php

```diff
@@ -34,7 +36,12 @@ public function build(ContainerBuilder $container)
 
         /** @var $extension SecurityExtension */
         $extension = $container->getExtension('security');
-        $extension->addSecurityListenerFactory(new OAuthFactory());
+
+        if (interface_exists(AuthenticatorFactoryInterface::class)) {
```

Comment:
This feels a bit hacky, so for future reference please add a comment to explain what's happening.